ambry-api/src/main/java/com/github/ambry/quota/Quota.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+import com.github.ambry.quota.storage.QuotaOperation;
+import java.util.Collections;
+import java.util.Set;
+
+
+/**
+ * The quota for a particular Ambry resource.
+ * @param <T> the type of the quota value.
+ */
+public class Quota<T> {
+  private final QuotaName quotaName;
+  private final T quotaValue;
+  private final QuotaResource quotaResource;
+  private final Set<QuotaOperation> quotaOperations;
```

Comment:
what is the purpose of quota operations here? Does it mean that a quota is only for certain operations? Like in a `QuotaSource`, we can get a list of `Quota`s, and they only works for Post operation?

Reply:
removed.